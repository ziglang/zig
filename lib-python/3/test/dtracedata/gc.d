python$target:::function-entry
/copyinstr(arg1) == "start"/
{
    self->trace = 1;
}

python$target:::gc-start,
python$target:::gc-done
/self->trace/
{
    printf("%d\t%s:%ld\n", timestamp, probename, arg0);
}

python$target:::function-return
/copyinstr(arg1) == "start"/
{
    self->trace = 0;
}
