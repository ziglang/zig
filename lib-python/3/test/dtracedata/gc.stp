global tracing

probe process.mark("function__entry")
{
    funcname = user_string($arg2);

    if (funcname == "start") {
        tracing = 1;
    }
}

probe process.mark("gc__start"), process.mark("gc__done")
{
    if (tracing) {
        printf("%d\t%s:%ld\n", gettimeofday_us(), $$name, $arg1);
    }
}

probe process.mark("function__return")
{
    funcname = user_string($arg2);

    if (funcname == "start") {
        tracing = 0;
    }
}
