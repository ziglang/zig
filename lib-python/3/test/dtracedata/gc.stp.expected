gc__start:0
gc__done:0
gc__start:1
gc__done:0
gc__start:2
gc__done:0
gc__start:2
gc__done:1
